source/uri/uris.adb

```diff
@@ -104,7 +104,15 @@ package body URIs is
          --  Check if we have path in form of UNC:  \\host\share\path
 
          if Found (0) = GNAT.Regpat.No_Match then
-            Add_All_Paths (URI, Full_Path, "");
+            if Full_Path (Full_Path'Last)
```

Comment:
I would use this way
```ada
Real_Full_Path : constant String := Ada.Directories.Full_Path (Full_Path);
begin
   return Real_From_File (Real_Full_Path);
end From_File;
```
And copy a From_File body to Real_From_File function.

This way we avoid using directory separator comparison. And this will work for several slashes at the end (like `/tmp/path///`).

Reply:
Good idea, done!